src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -202,6 +204,18 @@ public int getInstanceCap() {
         return instanceCap;
     }
 
+    public void setSlaveConnectTimeout(int slaveConnectTimeout) {
+        if (slaveConnectTimeout < 0) {
+            this.slaveConnectTimeout = 0;
+        } else {
+            this.slaveConnectTimeout = slaveConnectTimeout;
+        }
+    }
+
+    public int getSlaveConnectTimeout() {
+        return slaveConnectTimeout;
```

Comment:
when upgrading slaveConnectTimeout  won't be in the object, so it will get set to 0 IIRC 
You need to handle it in `readResolve` to set it to the default, or here to return the default if it is 0